cmd/gossamer/README.md

```diff
@@ -0,0 +1,127 @@
+    keystore
+- `import-runtime` - This subcommand takes a [Wasm runtime binary](https://wiki.polkadot.network/docs/learn-wasm) and
+  uses it to generate a [genesis](https://wiki.polkadot.network/docs/glossary#genesis) configuration file; it does not
+  require any flags, but expects the path to a Wasm file to be provided as a command-line parameter (example:
+  `./bin/gossamer import-runtime runtime.wasm > genesis.json`).
+- `build-spec` - This subcommand allows the user to "compile" a human-readable Gossamer genesis configuration file into
+  a format that the Gossamer node can consume. If the `--genesis` parameter is not provided, the generated genesis
+  configuration will represent the Gossamer default configuration.
+  - `genesis` - path to the human-readable configuration file that should be compiled into a format that Gossamer can
+    consume
+- `import-state` - The `import-state` subcommand allows a user to seed [Gossamer storage](../../dot/state) with
+  key-value pairs in the form of a JSON file. The input for this subcommand can be retrieved from
+  [the state_getPairs RPC endpoint](https://github.com/w3f/PSPs/blob/master/PSPs/drafts/psp-6.md#1114-state_getpairs).
+  - `first-slot` - the first [BABE](https://wiki.polkadot.network/docs/learn-consensus#block-production-babe) slot,
+    which is the timestamp of the first block divided by the slot duration
+  - `header` - path to a JSON file that describes the block header corresponding to the given state
+  - `state` - path to a JSON file that contains the key-value pairs with which to seed Gossamer storage
+- `init` - This subcommand accepts a genesis configuration file and uses it to initialise the Gossamer node and its
+  state.
+  - `genesis` - path to the "compiled" genesis configuration file that should be used to initialise the Gossamer node
+    and its state
+- `export` - The `export` subcommand transforms a genesis configuration and Gossamer state into a TOML configuration
+  file.
+  - `config` - path to a TOML configuration file (e.g. [Polkadot](../../chain/polkadot/config.toml))
+  - `basepath` - path to the Gossamer data directory that defines the state to export
+
+## Client Components
+
+In its default method of execution, Gossamer orchestrates a number of modular services that run
+[concurrently as goroutines](https://www.golang-book.com/books/intro/10) and work together to implement the protocols of
+a blockchain network. Alongside these services, Gossamer manages [a keystore](#keystore), [a runtime](#runtime), and
+[a telemetry service](#telemetry), all of which are described in greater detail below. The entry point to the Gossamer
+blockchain client capabilities is the `gossamerAction` function that is defined in [main.go](main.go), which in turn
+invokes the `NewNode` function in [dot/node.go](../../dot/node.go). `NewNode` calls into functions that are defined in
```

Comment:
Just wanted to revisit this comment and make an amendment - once we have a better release/tagging system I am _all for_ linking to line numbers in tagged releases ✅

Reply:
The suggestion links to a certain commit + line number, so it won't change.
The problem with linking to a file (even without line number) is that it might move and then the document becomes broken. At least link to the commit (and remove the line number)? So if we decide to ahem split packages, it won't be broken 😉

Reply:
If the code changes we need to update the documentation to match. We should have CI jobs that check our docs for broken links.

Reply:
Create an issue for that 😄 👍

Reply:
Added something to the Team Code Review doc ✅